widgets/src/image.rs

```diff
@@ -101,21 +102,30 @@ impl Image {
 }
 
 impl ImageRef {
-    pub fn load_image_dep_by_path(&self, cx: &mut Cx, image_path: &str) {
+    /// Loads the image at the given `image_path` into this `ImageRef`.
+    pub fn load_image_dep_by_path(&self, cx: &mut Cx, image_path: &str) -> Result<(), ImageError> {
         if let Some(mut inner) = self.borrow_mut() {
             inner.load_image_dep_by_path(cx, image_path)
+        } else {
+            Ok(()) // preserving existing behavior of silent failures.
```

Comment:
i think @makepaddev  had some argument for not having an `Result` in the rendering part of the api, but from my point of view - it's better to fail sooner rather than debug silent errors. + it would remove huge amount of unwraps from the codebase.